R/value.R

```diff
@@ -107,3 +107,275 @@ inspect_na_value <- function(value) {
   return(success())
 
 }
+
+
+
+#' Check if the passed entity is a single floating point which is (strictly)
+#' greater than another specified value.
+#'
+#' @param value the value to check
+#' @param comparator the minimum allowed value, inclusive or exclusive.
+#' @param strictly If TRUE (default), the comparator value is checked exclusive
+#' \code{(>)}. If FALSE the comparator value is checked inclusive \code{(>=)}.
+#'
+#' @param allow_na if true, accept a value that is NA.
+#' @param allow_null if true, accept a value that is NULL.
+#'
+#' @examples
+#' \dontrun{
+#' # For assertion
+#' assertthat::assert_that(qscheck::is_greater_than_value(value, comparator))
+#' # For check
+#' if (qscheck::is_greater_than_value(value, comparator)) {}
+#' }
+#'
+#' @concept value
+#' @export
+is_greater_than_value <- function(
+  value, comparator, strictly = TRUE,
+  allow_na = FALSE, allow_null = FALSE
+) {
+
+  res <- inspect_greater_than_value(
+    value, comparator,
+    strictly = strictly,
+    allow_na = allow_na,
+    allow_null = allow_null
+  )
+
+  return(res$valid)
+}
+assertthat::on_failure(is_greater_than_value) <- function(call, env) {
+  value <- callget(call, env, "value", NULL)
+  comparator <- callget(call, env, "comparator", NULL)
+  strictly <- callget(call, env, "strictly", TRUE)
+  allow_na <- callget(call, env, "allow_na", FALSE)
+  allow_null <- callget(call, env, "allow_null", FALSE)
+
+  res <- inspect_greater_than_value(
+    value, comparator,
+    strictly = strictly,
+    allow_na = allow_na,
+    allow_null = allow_null
+  )
+
+if (is.null(value) || is.null(comparator)) {
```

Comment:
if you allow null for the value, now this will fail even if it's allowed. 

For the comparator, that's a different story.

Reply:
ah no we are in the on failure... 

Still, the reason will tell you that for the value. For the comparator, it's a different story. Qscheck always assumes (incorrectly) that the parameters other than the one to monitor are correct. No checks are performed if they are integers, for example, or null. This is a limitation, and it's a limitation I artificially left there simply because of lack of time.

So in principle, you should have the check in the inspect as well, and let this information in the reason. But... at this point the reason is not related to the parameter you are checking, but to something that the programmer has messed up, hence why I postponed introducing it. in any case, you should return false also in the checking function, otherwise it gets desynchronised.